test/contract.test.ts

```diff
@@ -761,4 +797,35 @@ describe('contract', function () {
     })
     expect(BigInt(state2.asset.alphAmount)).toEqual(MINIMAL_CONTRACT_DEPOSIT)
   })
+
+  it('should get the contract bytecode for testing', async () => {
+    expect(Add.contract.publicFunctions().length).not.toEqual(Add.contract.functions.length)
+    const instance0 = (await Add.deploy(signer, { initialFields: { sub: randomContractId(), result: 0n } }, true))
+      .contractInstance
+    const state0 = await instance0.fetchState()
+    expect(state0.bytecode).toEqual(Add.contract.getByteCodeForTesting())
+    expect(state0.bytecode).not.toEqual(Add.contract.bytecode)
+    expect(state0.bytecode).not.toEqual(Add.contract.bytecodeDebug)
+    expect(state0.codeHash).not.toEqual(Add.contract.codeHash)
+    expect(state0.codeHash).not.toEqual(Add.contract.codeHashDebug)
+    expect(Add.contract.hasCodeHash(state0.codeHash)).toEqual(true)
+
+    expect(Assert.contract.publicFunctions().length).toEqual(Assert.contract.functions.length)
```

Comment:
This should return the correct public function length instead of the length for testing contracts

Reply:
We generate the testing bytecode only if the `exposePrivateFunctions` is true, and if all functions are public, we will use the `bytecodeDebug` directly. The purpose of this line of code is to check all functions are public.